Improve Payment Templates: Complete, Default, and Resolve Variables

cestercian opened the pull request:
## Type of Change
<!-- Put an `x` in the boxes that apply -->

- [x] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
<!-- Describe your changes in detail -->

This PR addresses several issues in the connector template files that could cause runtime errors or compilation failures when the templates are used to implement new payment connectors:

1. In `connector-template/mod.rs`:
   - Replaced `todo!()` in `get_currency_unit()` with a default implementation to prevent runtime panics
   - Fixed template variable in `base_url()` method to properly include the `downcase` filter
   - Added implementations for empty `AccessTokenAuth` interface methods

2. In `connector-template/test.rs`:
   - Implemented test helper functions that were previously returning `None`
   - Added realistic test data for payment methods and payment information
   - Added necessary imports for the new functionality

3. In `connector-template/transformers.rs`:
   - Added missing import for the `cards` module which is used for the `CardNumber` type

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
<!--
Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
-->

These template files serve as a starting point for implementing new payment connectors. The issues fixed in this PR would cause confusion or errors for developers trying to use these templates. By providing working defaults and fixing structural issues, we make the connector implementation process more straightforward and avoid the following problems:

1. Runtime panics from `todo!()` calls
2. Compilation errors due to missing imports
3. Template rendering issues due to inconsistent variable formatting
4. Test failures due to `None` values in test helpers

## How did you test it?
<!--
Did you write an integration/unit/API test to verify the code changes?
Or did you test this change manually (provide relevant screenshots)?
-->

The template files were manually reviewed to ensure consistent formatting and proper default implementations. Since these are template files, the changes ensure that when an actual connector is implemented using these templates, the resulting code will compile and run without immediate errors.

## Checklist
<!-- Put an `x` in the boxes that apply -->

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

deepanshu-iiitu commented on connector-template/mod.rs:
```diff
@@ -169,6 +169,57 @@ impl
+
+    fn get_url(
+        &self,
+        _req: &RouterData<AccessTokenAuth, AccessTokenRequestData, AccessToken>,
+        _connectors: &Connectors,
+    ) -> CustomResult<String, errors::ConnectorError> {
+        Err(errors::ConnectorError::NotImplemented("get_url method".to_string()).into())
+    }
+
+    fn get_request_body(
+        &self,
+        _req: &RouterData<AccessTokenAuth, AccessTokenRequestData, AccessToken>,
+        _connectors: &Connectors,
+    ) -> CustomResult<RequestContent, errors::ConnectorError> {
+        Err(errors::ConnectorError::NotImplemented("get_request_body method".to_string()).into())
+    }
+
+    fn build_request(
+        &self,
+        _req: &RouterData<AccessTokenAuth, AccessTokenRequestData, AccessToken>,
+        _connectors: &Connectors,
+    ) -> CustomResult<Option<Request>, errors::ConnectorError> {
+        Err(errors::ConnectorError::NotImplemented("build_request method".to_string()).into())
+    }
+
+    fn handle_response(
+        &self,
+        _data: &RouterData<AccessTokenAuth, AccessTokenRequestData, AccessToken>,
+        _res: Response,
+    ) -> CustomResult<RouterData<AccessTokenAuth, AccessTokenRequestData, AccessToken>, errors::ConnectorError> {
+        Err(errors::ConnectorError::NotImplemented("handle_response method".to_string()).into())
+    }
+
+    fn get_error_response(
+        &self,
+        _res: Response,
+        _event_builder: Option<&mut ConnectorEvent>
+    ) -> CustomResult<ErrorResponse, errors::ConnectorError> {
+        Err(errors::ConnectorError::NotImplemented("get_error_response method".to_string()).into())
+    }
```
Access token flow is not needed in template as very few payment processors require this flow.
Please revert these changes.

cestercian replied:
Done 🎉 !

deepanshu-iiitu commented on connector-template/transformers.rs:
```diff
@@ -2,6 +2,7 @@ use common_enums::enums;
 use serde::{Deserialize, Serialize};
 use masking::Secret;
 use common_utils::types::{StringMinorUnit};
+use hyperswitch_domain_models::payment_method_data::cards;
```
This can be removed

cestercian replied:
sure!

cestercian commented:
Hey @deepanshu-iiitu,
Thank you for your review and feedback! I've made the changes you requested:
 - Reverted the AccessToken implementation to keep it empty as in the original template
 - Removed the unnecessary cards import from transformers.rs.
 
Let me know if you'd like to see any other modifications to the PR!

cestercian commented:
Hey @deepanshu-iiitu,  

Could you take a look at this PR? It seems like it might have been missed. Thanks!
